### PR TITLE
[Wrangler] Update the commands list

### PR DESCRIPTION
**Pull request by lrapoport-cf**
## What this PR solves / how to test:
This PR cleans up and standardises the look and feel of all `wrangler` commands as displayed by `wrangler --help` and `wrangler <cmd> --help`. 

<details>
<summary>Please refer to original designs here</summary>
Screenshot 2024-06-19 at 20 38 13
</details>

**Before**
Screenshot 2024-02-26 at 18 50 41

**After**
Screenshot 2024-06-19 at 20 32 12

Here is the list of all public facing wrangler commands, in their updated version:

<details>
<summary>**wrangler deployments --help**</summary>
Screenshot 2024-06-19 at 19 58 58
</details>

<details>
<summary>**wrangler deployments --x-versions --help**</summary>
Screenshot 2024-06-19 at 19 59 30
</details>

<details>
<summary>**npx wrangler rollback --help**</summary>
Screenshot 2024-06-19 at 20 12 21
</details>

<details>
<summary>**npx wrangler rollback --x-versions --help**</summary>
Screenshot 2024-06-19 at 20 13 05
</details>

<details>
<summary>**npx wrangler versions --x-versions --help**</summary>
Screenshot 2024-06-19 at 20 14 33
</details>

<details>
<summary>**wrangler triggers --x-versions --help**</summary>
Screenshot 2024-06-19 at 20 15 36
</details>

<details>
<summary>**wrangler kv --help**</summary>
Screenshot 2024-06-19 at 20 18 59
</details>

<details>
<summary>**wrangler queues --help**</summary>
Screenshot 2024-06-19 at 20 18 25
</details>

<details>
<summary>**wrangler r2 --help**</summary>
Screenshot 2024-06-19 at 20 19 31
</details>

<details>
<summary>**wrangler d1 --help**</summary>
Screenshot 2024-06-19 at 20 21 44
</details>

<details>
<summary>**wrangler vectorize --help**</summary>
Screenshot 2024-06-19 at 20 22 49
</details>

<details>
<summary>**wrangler hyperdrive --help**</summary>
Screenshot 2024-06-19 at 20 23 34
</details>

<details>
<summary>**wrangler pages --help**</summary>
Screenshot 2024-06-19 at 20 24 30
</details>

<details>
<summary>**wrangler mtls-certificate --help**</summary>
Screenshot 2024-06-19 at 20 25 22
</details>

<details>
<summary>**wrangler cloudchamber --help**</summary>
Screenshot 2024-06-19 at 20 26 20
</details>

<details>
<summary>**wrangler pubsub --help**</summary>
Screenshot 2024-06-19 at 20 27 38
</details>

<details>
<summary>**wrangler dispatch-namespace --help**</summary>
Screenshot 2024-06-19 at 20 28 30
</details>

<details>
<summary>**wrangler ai --help**</summary>
Screenshot 2024-06-19 at 20 29 44
</details>

<details>
<summary>**wrangler login --help**</summary>
Screenshot 2024-06-19 at 20 30 29
</details>

## Author has addressed the following

- Tests
  - [ ] TODO (before merge)
  - [x] Included
  - [ ] Not necessary because:
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [ ] TODO (before merge)
  - [x] Included
  - [ ] Not necessary because:
- Public documentation
  - [ ] TODO (before merge): put up PR to update wrangler commands list to match changes in this PR (a la this PR put up initially but which was closed because went stale: https://github.com/cloudflare/cloudflare-docs/pull/9430 )
  - [x] Cloudflare docs PR(s): https://github.com/cloudflare/cloudflare-docs/pull/11514
  - [ ] Not necessary because: simple bug fix

### CODE REVIEW

File: packages/wrangler/src/index.ts [hunk 1]
```diff
@@ -290,114 +300,189 @@ export function createCLIParser(argv: string[]) {
...
+		"📖Open wrangler commands doc in your browser",
 		docsOptions,
 		docsHandler
 	);
 
 	// init
 	wrangler.command(
 		"init [name]",
-		"📥 Initialize a basic Worker project, including a wrangler.toml file",
+		"🔸Innitialize a basic worker application",
```
Comment (Skye-31) [inline packages/wrangler/src/index.ts:312]: Small typo
```suggestion
		"🔸Initialize a basic worker application",
```

File: packages/wrangler/src/index.ts [hunk 2]
```diff
@@ -290,114 +300,189 @@ export function createCLIParser(argv: string[]) {
...
-		"✨ Generate a new Worker project from an existing Worker template. See https://github.com/cloudflare/templates",
-		generateOptions,
-		generateHandler
-	);
-
-	// [DEPRECATED] build
-	wrangler.command("build", false, buildOptions, buildHandler);
-
-	// [DEPRECATED] config
-	wrangler.command("config", false, noOpOptions, configHandler);
```
Comment (Skye-31) [inline packages/wrangler/src/index.ts:319]: Removing these outside of a major version is still a breaking change, regardless of whether they were previously deprecated, isn't it?

File: packages/wrangler/src/index.ts (hunk 1 above)
Comment (lrapoport-cf) [reply packages/wrangler/src/index.ts]: thanks!

**Comment by lrapoport-cf** [comment]
@RamIdeas thanks for the feedback! the emoji changes are based on a proposal from the design team -- i'll get them looped into the discussion. for my two cents, i agree that the consistency has a lower fun quotient 😄, but i do think it adds some informational value that the more assorted collection lacks: it signals that there is a loose relationship between the items in the different emoji groups, and also distracts less from the command descriptions.

**Comment by lrapoport-cf** [comment]
cc @Meahaa

**Comment by Meahaa** [comment]
Excellent points let's:

1. Add the space
2. Keep the emoji's agree it's more fun to have them and we can take a pass in the future to make sure they actually make sense
3. I'm not following too clearly here - I liked the orange text because it's a nice brand call out

**Comment by RamIdeas** [comment]
> 3. I'm not following too clearly here - I liked the orange text because it's a nice brand call out

I think you're referring to the "open beta" text in the figma design doc which I think being on-brand orange is great!

I was referring to the ⚑ (flag?) emoji which blends into the text a little – a non-exhaustive set of alternatives to consider:
- 🧪 test tube
- 👩‍🔬 scientist
- 🥼 lab coat
- 🧫 petri dish
- 🔬 microscope

**Comment by petebacondarwin** [comment]
@Meahaa - can you take a look at @RamIdeas' last comment? Should we update that last emoji?
@lrapoport-cf - can we rebase and fix the conflicts while updating to fit the agreed points between @RamIdeas and @Meahaa ?

**Comment by Meahaa** [comment]
ooh I like the test tube let's go with that!

**Comment by CarmenPopoviciu** [comment]
> Excellent points let's:
> 
> 1. Add the space
> 2. Keep the emoji's agree it's more fun to have them and we can take a pass in the future to make sure they actually make sense
> 3. I'm not following too clearly here - I liked the orange text because it's a nice brand call out

@Meahaa I am taking over this PR to cross it over the finish line. Can you please clarify point #2 for me? Do we keep the current emojis as they are, or do we change them as per design?

**Comment by CarmenPopoviciu** [comment]
@Meahaa @RamIdeas @petebacondarwin I've added a before and after screenshot to the description of this PR. Before reviewing the code, I would like to get consensus that, in terms of visuals, things are as we want them to be. TY! <3

**Comment by Meahaa** [comment]
Yup! Let's just keep the current emojis for now! Agree with the fact that they make them more fun for now and we can always update to make them even better late!

File: packages/wrangler/src/versions/deployments/index.ts [hunk 3]
```diff
@@ -18,13 +18,13 @@ export default function registerVersionsDeploymentsSubcommands(
 	versionDeploymentsYargs
 		.command(
 			"list",
-			"Displays the 10 most recent deployments of your Worker [beta]",
```
Comment (CarmenPopoviciu) [inline packages/wrangler/src/versions/deployments/index.ts:21]: removing as the `beta` nature of these commands is signalled at command level, which should be enough?

File: packages/wrangler/src/index.ts [hunk 4]
```diff
@@ -249,13 +256,24 @@ export function createCLIParser(argv: string[]) {
...
+		})
+		.epilogue(
+			`Please report any issues to ${highlight(
+				"https://github.com/cloudflare/workers-sdk/issues/new/choose",
+				"#3B818D"
+			)}`
+		);
 
+	wrangler.updateStrings({
+		"Commands:": `${embolden("COMMANDS")}`,
```
Comment (CarmenPopoviciu) [inline packages/wrangler/src/index.ts:268]: this should have been smth like
```
wrangler.getLocale().getValue("Commands")
```
but `yargs` doesn't support anything near that, so here we are ¯\_(ツ)_/¯

File: packages/wrangler/src/index.ts [hunk 5]
```diff
@@ -204,33 +210,34 @@ export function createCLIParser(argv: string[]) {
 		})
 		.scriptName("wrangler")
 		.wrap(null)
+		.locale("en_US")
```
Comment (CarmenPopoviciu) [inline packages/wrangler/src/index.ts:213]: forcing locale here, since we don't support anything other than `en`. This also ensures we don't run into any unexpected behaviour when we call `wrangler.updateStrings()` further down

File: packages/wrangler/src/utils/stdout-styling.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+ * not support colors, it will return the unchanged text instead.
+ */
+export function highlight(text: string, colorHex: string): string {
+	return supportsColor.stdout ? chalk.hex(colorHex)(text) : text;
+}
+
+/**
+ * Terminal styling util function that makes text bold.
+ */
+export function embolden(text: string): string {
```
Comment (CarmenPopoviciu) [inline packages/wrangler/src/utils/stdout-styling.ts:15]: happy for other name suggestions

File: packages/wrangler/src/versions/deployments/index.ts (hunk 3 above)
Comment (CarmenPopoviciu) [reply packages/wrangler/src/versions/deployments/index.ts:21]: cc// @RamIdeas

File: packages/wrangler/src/utils/stdout-styling.ts [hunk 7]
```diff
@@ -0,0 +1,17 @@
+import chalk from "chalk";
+import supportsColor from "supports-color";
+
+/**
+ * Terminal styling util function that sets text color. If terminal does
+ * not support colors, it will return the unchanged text instead.
+ */
+export function highlight(text: string, colorHex: string): string {
+	return supportsColor.stdout ? chalk.hex(colorHex)(text) : text;
```
Comment (penalosa) [inline packages/wrangler/src/utils/stdout-styling.ts:9]: Doesn't chalk check for this internally anyway?

File: packages/wrangler/src/utils/stdout-styling.ts (hunk 6 above)
Comment (penalosa) [reply packages/wrangler/src/utils/stdout-styling.ts:15]: What about just importing `bold` from `chalk` and using that directly? Is this wrapper needed?

File: packages/wrangler/src/index.ts [hunk 8]
```diff
@@ -204,33 +210,34 @@ export function createCLIParser(argv: string[]) {
...
 			alias: "c",
-			describe: "Path to .toml configuration file",
+			describe: "⚑ Path to .toml configuration file",
 			type: "string",
 			requiresArg: true,
 		})
 		.option("env", {
 			alias: "e",
-			describe: "Environment to use for operations and .env files",
+			describe: "⚑ Environment to use for operations and .env files",
```
Comment (penalosa) [inline packages/wrangler/src/index.ts:230]: This emoji seems like visual clutter for not much benefit, especially since it's only on global flags (not command flags)

File: packages/wrangler/src/index.ts (hunk 4 above)
Comment (penalosa) [reply packages/wrangler/src/index.ts:268]: I know this comes from the designs, but I really don't like this change. All caps bold seems far too significant here, since this shouldn't be the focus of the user (they should be focusing on the commands themselves). All caps text is also usually harder to read, I find.

File: packages/wrangler/src/index.ts [hunk 9]
```diff
@@ -249,13 +256,24 @@ export function createCLIParser(argv: string[]) {
...
 
+	wrangler.updateStrings({
+		"Commands:": `${embolden("COMMANDS")}`,
+		"Options:": `${embolden("OPTIONS")}`,
+		"Positionals:": `${embolden("POSITIONALS")}`,
+	});
 	wrangler.group(
 		["experimental-json-config", "config", "env", "help", "version"],
-		"Flags:"
+		`${embolden("GLOBAL FLAGS")}`
```
Comment (penalosa) [inline packages/wrangler/src/index.ts:274]: As above

File: packages/wrangler/src/index.ts [hunk 10]
```diff
@@ -290,103 +308,208 @@ export function createCLIParser(argv: string[]) {
...
+	 */
+	/******************************************************/
+	/*                 WRANGLER COMMANDS                  */
+	/******************************************************/
 	// docs
 	wrangler.command(
-		"docs [command..]",
-		"📚 Open wrangler's docs in your browser",
+		"docs [command]",
+		"📚 Open the wrangler commands documentation in your browser\n",
```
Comment (penalosa) [inline packages/wrangler/src/index.ts:332]:
```suggestion
		"📚 Open Wrangler's command documentation in your browser\n",
```

File: packages/wrangler/src/index.ts [hunk 11]
```diff
@@ -290,103 +308,208 @@ export function createCLIParser(argv: string[]) {
...
-	wrangler.command("build", false, buildOptions, buildHandler);
-
-	// [DEPRECATED] config
-	wrangler.command("config", false, noOpOptions, configHandler);
-
 	// dev
 	wrangler.command(
 		"dev [script]",
-		"👂 Start a local server for developing your worker",
+		"👂 Start a local server for developing a Worker",
```
Comment (penalosa) [inline packages/wrangler/src/index.ts:349]: Why this change? It's not just `a` Worker, it's specifically `your` worker in the current directory

File: packages/wrangler/src/index.ts [hunk 12]
```diff
@@ -617,7 +689,7 @@ export function createCLIParser(argv: string[]) {
 	// whoami
 	wrangler.command(
 		"whoami",
-		"🕵️  Retrieve your user info and test your auth config",
+		"🕵️  Retrieve your user information and test your authentication configuration",
```
Comment (penalosa) [inline packages/wrangler/src/index.ts:692]: This is pretty long for the terminal now—how about just `🕵️ Retrieve your user information`?

File: packages/wrangler/src/index.ts (hunk 4 above)
Comment (petebacondarwin) [reply packages/wrangler/src/index.ts:268]: I think this is quite subjective. One could argue that making it all-caps (since it might be more difficult to read) does in fact focus the users attention on that actual commands themselves.

File: packages/wrangler/src/index.ts (hunk 8 above)
Comment (petebacondarwin) [reply packages/wrangler/src/index.ts:230]: I agree. Did that also come from the designs? Would be happy if this was not added.

File: packages/wrangler/src/utils/stdout-styling.ts (hunk 7 above)
Comment (CarmenPopoviciu) [reply packages/wrangler/src/utils/stdout-styling.ts:9]: no idea. I'll check

File: packages/wrangler/src/index.ts (hunk 8 above)
Comment (CarmenPopoviciu) [reply packages/wrangler/src/index.ts:230]: yeah, I 100% agree with you! I'm going to go ahead and break design here

File: packages/wrangler/src/index.ts (hunk 11 above)
Comment (CarmenPopoviciu) [reply packages/wrangler/src/index.ts:349]: no idea. The wording came from design, but I'll revert back to `your`

File: packages/wrangler/src/index.ts (hunk 12 above)
Comment (CarmenPopoviciu) [reply packages/wrangler/src/index.ts:692]: SGTM 👍

File: packages/wrangler/src/index.ts (hunk 4 above)
Comment (CarmenPopoviciu) [reply packages/wrangler/src/index.ts:268]: interesting. I feel the exact opposite 😅 . I find 
Screenshot 2024-06-20 at 09 56 02

much clearer and easier to scan than
Screenshot 2024-06-20 at 09 55 33

In the first option, I actually can immediately focus on the commands themselves, as it is 100% clear what I'm looking at (the grouping titles in caps make that very clear), as opposed to the second option, where I have to spend a min to process the grouping, as there is nothing that really stands out. Even if those headings were in bold, they still tend to get lost in the overall wall of text we output. But again, that's how my brain works, so this is a very personal preference

To be clear, the original design didn't use the caps. This is smth Arun and I talked abt and agreed might be a change for the better, as we felt it makes the grouping clearer and easier to scan. We've also been looking into CLI design systems and drew inspiration from https://clig.dev/#help.

Having said that, if folks feel strong against this change, like you do, happy to convert those headings back to small caps

File: packages/wrangler/src/utils/stdout-styling.ts (hunk 6 above)
Comment (CarmenPopoviciu) [reply packages/wrangler/src/utils/stdout-styling.ts:15]: yeah, since we don't need `highlight` anymore either, I'll remove this too

File: packages/wrangler/src/index.ts (hunk 4 above)
Comment (penalosa) [reply packages/wrangler/src/index.ts]: Sounds good! I'll retract this comment then

Comment (CarmenPopoviciu) [reply packages/wrangler/src/index.ts]: it was a totally valid comment to have <3!!

**Review by petebacondarwin — APPROVED** [review]
